### PR TITLE
Use smart pointers for RemoteAuthentication in PcapRemoteDeviceList.

### PR DESCRIPTION
**Pull request by Dimi1010**
Part of #1431. Implements point 3.

### CODE REVIEW

File: Pcap++/src/PcapRemoteDevice.cpp [hunk 1]
```diff
@@ -19,14 +19,14 @@ pcap_rmtauth PcapRemoteAuthentication::getPcapRmAuth() const
 	return result;
 }
 
-PcapRemoteDevice::PcapRemoteDevice(pcap_if_t* iface, PcapRemoteAuthentication* remoteAuthentication, const IPAddress& remoteMachineIP, uint16_t remoteMachinePort)
+PcapRemoteDevice::PcapRemoteDevice(pcap_if_t* iface, std::shared_ptr<PcapRemoteAuthentication> remoteAuthentication, const IPAddress& remoteMachineIP, uint16_t remoteMachinePort)
 	: PcapLiveDevice(iface, false, false, false)
+	, m_RemoteMachineIpAddress(remoteMachineIP)
```
Comment (tigercosmos) [inline Pcap++/src/PcapRemoteDevice.cpp:24]: recommend the reorder of the construction list by the order of the arguments.

**Review by tigercosmos — APPROVED** [review]

File: Pcap++/src/PcapRemoteDevice.cpp (hunk 1 above)
Comment (Dimi1010) [reply Pcap++/src/PcapRemoteDevice.cpp:24]: It is currently ordered the way the members are declared in the class, which is the actual way they will be initialized.

Comment (tigercosmos) [reply Pcap++/src/PcapRemoteDevice.cpp:24]: I see

File: Pcap++/src/PcapRemoteDevice.cpp
Comment (seladb) [inline Pcap++/src/PcapRemoteDevice.cpp:1]: There are a few more occurrences of `NULL`, maybe we should replace them with `nullptr`?

**Review by seladb — APPROVED** [review]
Please see one comment, otherwise LGTM

File: Pcap++/src/PcapRemoteDevice.cpp
Comment (Dimi1010) [reply Pcap++/src/PcapRemoteDevice.cpp:1]: Added 501bda6. There are some C-style casts left over, but those are handled in #1443.

**Pull request merged**